include/triton/Conversion/TritonGPUToLLVM/Utility.h

```diff
@@ -736,7 +736,7 @@ isSimpleSharedMemoryAccess(ArrayRef<int64_t> shape,
   auto rank = shape.size();
   auto swizzledLayout =
       dyn_cast<triton::gpu::SwizzledSharedEncodingAttr>(sharedEnc);
-  bool noSwizzling = swizzledLayout && swizzledLayout.getMaxPhase() == 1;
+  bool noSwizzling = !swizzledLayout || swizzledLayout.getMaxPhase() == 1;
```

Comment:
!swizzledLayout doesn't mean no swizzling, you have to check that the swizzling byte size is 0 too